### PR TITLE
test(auth-server): coverage for updatePayment

### LINKED ISSUE #4224: Code coverage for Direct Stripe Route - Update Payment
File: fxa-auth-server/lib/routes/subscriptions.js

Function: DirectStripeRoutes.updatePayment

Add coverage for above function. If any refactoring occurs as a part of coverage, ensure that any added functions also have full test coverage.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-1173)
┆Issue Number: FXA-1173

### PR DESCRIPTION
**Pull request by jackiemunroe**
- added code coverage for DirectStripeRoutes.updatePayment

fixes #4224

### CODE REVIEW

File: packages/fxa-auth-server/test/local/routes/subscriptions.js [hunk 1]
```diff
@@ -2319,7 +2319,92 @@ describe('DirectStripeRoutes', () => {
...
   });
 
-  describe('updatePayment', () => {});
+  describe('updatePayment', () => {
+    let request;
+    const paymentToken = 'tok_visa';
+
+    beforeEach(() => {
+      request = Object.create(VALID_REQUEST);
+      request.payload = { paymentToken };
```
Comment (chenba) [inline packages/fxa-auth-server/test/local/routes/subscriptions.js]: Any reason you favor `Object.create` over something like `request = {...VALID_REQUEST, payload: {paymentToken}};`?

File: packages/fxa-auth-server/test/local/routes/subscriptions.js [hunk 2]
```diff
@@ -2319,7 +2319,92 @@ describe('DirectStripeRoutes', () => {
...
+          )
+        );
+      });
+    });
+
+    describe('when the customer does not exist', () => {
+      it('throws an error', () => {
+        directStripeRoutesInstance.stripeHelper.fetchCustomer.returns();
+
+        return directStripeRoutesInstance.updatePayment(request).then(
```
Comment (chenba) [inline packages/fxa-auth-server/test/local/routes/subscriptions.js]: Mixing `async/await` in the previous test and promises here is kind of weird.  Can you also using `async/await` here, and `try/catch` the error you want to test against?

**Comment by chenba** [comment]
nit: subject of commit message doesn't conform to https://github.com/mozilla/fxa/blob/master/CONTRIBUTING.md#subject

File: packages/fxa-auth-server/test/local/routes/subscriptions.js (hunk 1 above)
Comment (jackiemunroe) [reply packages/fxa-auth-server/test/local/routes/subscriptions.js]: No reason in particular, it was just where my mind went. Updating.

File: packages/fxa-auth-server/test/local/routes/subscriptions.js (hunk 2 above)
Comment (jackiemunroe) [reply packages/fxa-auth-server/test/local/routes/subscriptions.js]: I can do that. I saw errors tested different ways in the code base when it came to promises - `try/catch` and `.then(()=>{}, err=>{})`. It didn't seem that we had a preferred methodology for error testing.

**Comment by jackiemunroe** [comment]
> nit: subject of commit message doesn't conform to https://github.com/mozilla/fxa/blob/master/CONTRIBUTING.md#subject

Updated commit subject

**Comment by chenba** [comment]
> > nit: subject of commit message doesn't conform to https://github.com/mozilla/fxa/blob/master/CONTRIBUTING.md#subject
> 
> Updated commit subject

Oh man, sorry I wasn't specific, so now I'm gonna sound super extra nit...  my comment was mainly about the phrasing of the commit message.  Although now that I've re-read the first bullet point on the contributing doc, it's a bit unclear if it's asking for the same thing, which is: start with a verb.

With that, the commit message would read "[implicit subject of patch/commit] does this thing".

We've been fairly but not completely consistent with that.

File: packages/fxa-auth-server/test/local/routes/subscriptions.js [hunk 3]
```diff
@@ -2319,7 +2319,91 @@ describe('DirectStripeRoutes', () => {
     });
   });
 
-  describe('updatePayment', () => {});
+  describe('updatePayment', () => {
+    let request;
+    const paymentToken = 'tok_visa';
+
+    beforeEach(() => {
+      request = request = { ...VALID_REQUEST, payload: { paymentToken } };
```
Comment (chenba) [inline packages/fxa-auth-server/test/local/routes/subscriptions.js]: Double assignment?

File: packages/fxa-auth-server/test/local/routes/subscriptions.js [hunk 4]
```diff
@@ -2319,7 +2319,91 @@ describe('DirectStripeRoutes', () => {
...
+
+      directStripeRoutesInstance.stripeHelper.updateCustomerPaymentMethod.returns();
+    });
+
+    describe('when the customer exists', () => {
+      it('updates the payment method', async () => {
+        const customer = Object.create(customerFixture);
```
Comment (chenba) [inline packages/fxa-auth-server/test/local/routes/subscriptions.js]: I think there's no need to new up an object here.  I don't see a reason why passing `customerFixture` on the next line wouldn't work.

File: packages/fxa-auth-server/test/local/routes/subscriptions.js (hunk 3 above)
Comment (jackiemunroe) [reply packages/fxa-auth-server/test/local/routes/subscriptions.js]: aha good catch... that's embarrassing ':D

File: packages/fxa-auth-server/test/local/routes/subscriptions.js (hunk 4 above)
Comment (jackiemunroe) [reply packages/fxa-auth-server/test/local/routes/subscriptions.js]: I can update. Thanks :)

**Comment by jackiemunroe** [comment]
> > > nit: subject of commit message doesn't conform to https://github.com/mozilla/fxa/blob/master/CONTRIBUTING.md#subject
> > 
> > 
> > Updated commit subject
> 
> Oh man, sorry I wasn't specific, so now I'm gonna sound super extra nit... my comment was mainly about the phrasing of the commit message. Although now that I've re-read the first bullet point on the contributing doc, it's a bit unclear if it's asking for the same thing, which is: start with a verb.
> 
> With that, the commit message would read "[implicit subject of patch/commit] does this thing".
> 
> We've been fairly but not completely consistent with that.

Ah. I see. Yeah, I read it as "if you use a verb in the subject, then it should use the imperative, present tense"

**Pull request merged**